sdk/src/main/java/com/bugsnag/android/Breadcrumbs.java

```diff
@@ -33,32 +38,25 @@ void clear() {
         store.clear();
     }
 
-    void setSize(int size) {
-        if (size < 0) {
-            Logger.warn("Ignoring invalid breadcrumb capacity. Must be >= 0.");
-            return;
-        }
-
-        this.maxSize = size;
-        // Remove oldest breadcrumbs until reaching the required size
-        while (store.size() > size) {
-            store.poll();
-        }
-    }
-
     private void addToStore(@NonNull Breadcrumb breadcrumb) {
         try {
             if (breadcrumb.payloadSize() > MAX_PAYLOAD_SIZE) {
                 Logger.warn("Dropping breadcrumb because payload exceeds 4KB limit");
                 return;
             }
             store.add(breadcrumb);
-            if (store.size() > maxSize) {
-                // Remove oldest breadcrumb
-                store.poll();
-            }
+            pruneBreadcrumbs();
         } catch (IOException ex) {
             Logger.warn("Dropping breadcrumb because it could not be serialized", ex);
         }
     }
+
+    private void pruneBreadcrumbs() {
+        int maxBreadcrumbs = configuration.getMaxBreadcrumbs();
+
+        // Remove oldest breadcrumb
```

Comment:
This comment just needs to be tweaked to state that the oldest breadcrumbs will be removed until the max size is reached, not just one breadcrumb